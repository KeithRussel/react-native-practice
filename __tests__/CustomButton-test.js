import React from 'react';
import {FlatList, Text} from 'react-native';
import renderer from 'react-test-renderer';
import CustomButton from '../src/utils/CustomButton';

test('renders correctly', () => {
  const tree = renderer.create(<CustomButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the Flatlist component', () => {
  const tree = renderer
    .create(
      <FlatList
        data={['Item1', 'Item2', 'Item3']}
        keyExtractor={item => item}
        renderItem={({item}) => <Text>{item}</Text>}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
