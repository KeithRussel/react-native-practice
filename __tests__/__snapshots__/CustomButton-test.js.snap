// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "height": 50,
        "marginBottom": 15,
        "width": 150,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  />
</View>
`;

exports[`renders the Flatlist component 1`] = `
<RCTScrollView
  data={
    Array [
      "Item1",
      "Item2",
      "Item3",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item1
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item2
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
